### PR TITLE
Somnia bot i18n translation

### PR DESCRIPTION
**Pull request by Godlander**
## Describe your changes

use i18n translations for ability descriptions

## Testing/validation

image

## Checklist before requesting a review (leave this PR as draft if any part of this list is not done.)

- [ ] I have commented my code in hard-to understand areas.
- [ ] I have made corresponding changes to README or wiki.
- [ ] For front-end changes, I have updated the corresponding English translations.
- [ ] I have run `yarn run mini-ci` locally to validate format and lint.
- [ ] If I have added a new library or app, I have updated the deployment scripts to ignore changes as needed

### CODE REVIEW

File: apps/somnia/src/commands/archive.ts [hunk 1]
```diff
@@ -171,29 +168,36 @@ export async function autocomplete(interaction: AutocompleteInteraction) {
...
+    if (!archive[subcommand].includes(id as WeaponKey))
+      throw 'invalid weapon name'
+    return await weaponArchive(id as WeaponKey, lang, arg)
   }
   //artifacts archive
   else if (subcommand === 'artifact') {
-    return artifactArchive(id as ArtifactSetKey, name, data)
+    if (!archive[subcommand].includes(id as ArtifactSetKey))
+      throw 'invalid artifact name'
+    return artifactArchive(id as ArtifactSetKey, lang)
```
Comment (nguyentvan7) [inline apps/somnia/src/commands/archive.ts:193]: missing await

File: apps/somnia/src/commands/archive/weapon.ts [hunk 2]
```diff
@@ -33,24 +34,43 @@ function getDropdown(id: string, refine: string) {
...
+  await i18nInstance.loadNamespaces(namespace)
   const msg: any = {}
+  let name: string = translate(namespace, 'name', lang)
   let text = ''
-  //weapon rarity color
-  const rarity = getWeaponStat(id).rarity
+  const stat = getWeaponStat(id)
+  //mainstat
+  const ascension = stat.ascensionBonus[stat.mainStat.type] ?? [0]
+  const mainstat =
```
Comment (nguyentvan7) [inline apps/somnia/src/commands/archive/weapon.ts:53]: Why not use Pando? Could be a future TODO as well i suppose, once we have more of the UI sheets and sheets done

**Comment by Godlander** [comment]
how the heck did i do that

File: apps/somnia/src/commands/archive/weapon.ts [hunk 3]
```diff
@@ -61,13 +90,20 @@ export function weaponArchive(
     //name and passive
     name += ` (R${refinedisplay[refine]})`
     text +=
-      `\n\n**${data.passiveName}:** ` +
-      Object.values(data.passiveDescription[refine]).join('\n')
+      '\n\n**' +
```
Comment (nguyentvan7) [inline apps/somnia/src/commands/archive/weapon.ts:93]: prob should refactor this and the other text+= into a readable function
`text += passiveNameAndDesc()` for example

**Review by nguyentvan7 — CHANGES_REQUESTED** [review]
Undo the change to numDisplay, the other 2 comments can be for next PR

**Review by nguyentvan7 — APPROVED** [review]

**Pull request merged**